autorest/version.go

```diff
@@ -18,11 +18,14 @@ var version string
 // Version returns the semantic version (see http://semver.org).
 func Version() string {
```

Comment:
I like the updates that you've made here, much more readable.

As long as we're at it, we should get rid of the race-condition that is possible here by updating this function to use [sync/once](https://golang.org/pkg/sync/#Once) instead of the current string check.

Feel free to ignore this comment if you'd like. I can make these changes at a later date.